In [1]:
import pandas as pd
url="https://bd29ee0e-54ab-4daa-9671-d153865d1620.usrfiles.com/ugd/bd29ee_9a0760878dc7470aaa032c2cfbce892b.csv"
df=pd.read_csv(url)
df.head()

Name      Surname  Age  Smokes  AreaQ  Alkhol  Result
0    John         Wick   35       3      5       4       1
1    John  Constantine   27      20      2       5       1
2  Camela     Anderson   30       0      5       2       0
3    Alex       Telles   28       0      8       1       0
4   Diego     Maradona   68       4      5       6       1

In [2]:
df.shape

(59, 7)

In [3]:
y=df[["Result"]]
y.value_counts()

Result
0         31
1         28
dtype: int64

In [17]:
X=df[["Age","Smokes","AreaQ","Alkhol"]]
X.head()

Age  Smokes  AreaQ  Alkhol
0   35       3      5       4
1   27      20      2       5
2   30       0      5       2
3   28       0      8       1
4   68       4      5       6

In [24]:
from sklearn import model_selection
from sklearn.model_selection import KFold
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score
from sklearn.metrics import confusion_matrix
from sklearn.model_selection import cross_val_predict
from sklearn.metrics import roc_auc_score
from sklearn.metrics import classification_report

In [22]:
from sklearn import svm
import numpy as np
model1=svm.SVC(C=5,kernel="linear",random_state=0)
scores=cross_val_score(model1,X,np.asarray(y).reshape(y.shape[0],),cv=2)
print("Accuracy: ",scores.mean(),"+/-",scores.std())

Accuracy:  0.9494252873563218 +/- 0.01609195402298852


In [23]:
y_pred = cross_val_predict(model1, X, np.asarray(y).reshape(y.shape[0],), cv=2)
print("Confusion matrix: \n",confusion_matrix(y,y_pred))

Confusion matrix: 
 [[30  1]
 [ 2 26]]


In [25]:
print("Classification report: \n",classification_report(y,y_pred))

Classification report: 
               precision    recall  f1-score   support

           0       0.94      0.97      0.95        31
           1       0.96      0.93      0.95        28

    accuracy                           0.95        59
   macro avg       0.95      0.95      0.95        59
weighted avg       0.95      0.95      0.95        59



In [7]:
X=df[["Smokes"]]

In [10]:
model2=svm.SVC(C=4,kernel="poly",gamma="auto",degree=2,random_state=0)
scores=cross_val_score(model2,X,np.asarray(y).reshape(y.shape[0],),cv=2)
print("Accuracy: ",scores.mean(),"+/-",scores.std())

Accuracy:  0.7293103448275862 +/- 0.02931034482758621


#### Smoking is by far the most important factor in predicting lung cancer. Using only "Smokes" as input, we could obtain a model that is 73% correct. In other words, just tell me whether someone smokes or not, I can tell whether that person will develop lung cancer or not and 73% of the time, I will be correct. 

#### If we include other factors such as age and alcohol usage, the model is 95% correct.